dapp/src/components/TransactionHistory.js

```diff
@@ -184,63 +183,28 @@ const TransactionHistory = ({ isMobile }) => {
   const setFilters = (filters) => {
     _setFilters(filters)
     setCurrentPage(1)
+  }
+
+  const historyQuery = useTransactionHistoryQuery(account)
+
+  const history = useMemo(
+    () => (historyQuery.isSuccess ? historyQuery.data : []),
+    [historyQuery.isSuccess, historyQuery.data]
+  )
+
+  const shownHistory = useMemo(() => {
```

Comment:
I have searched the repo and it didn't seem there was a use of `useMemo` - I assume the pattern of `useEffect` + `useState` was used instead.

Reply:
Yeah this is a better approach. I defaulted to a less clean/practical pattern and haven't used `useMemo` where appropriate. Thanks for pointing this out